docs/guide/credentials.rst

```diff
@@ -77,6 +79,66 @@ been configured with an IAM role.
 For more information, see `IAM Roles for Amazon EC2 <http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html>`_.
 
 
+.. _ecs_credentials:
+
+Using IAM roles for Amazon EC2 Container Service Tasks
+------------------------------------------------------
+
+*With IAM roles for Amazon ECS tasks, you can specify an IAM role that can be
+used by the containers in a task.* This provides a strategy for managing credentials
+for your applications to use, similar to the way that Amazon EC2 instance profiles
+provide credentials to EC2 instances.
+
+Instead of creating and distributing your AWS credentials to the containers or
+using the EC2 instance’s role, you can associate an IAM role with an ECS task definition or
+``RunTask`` `API <http://docs.aws.amazon.com/aws-sdk-php/v3/api/api-ecs-2014-11-13.html#runtask>`_ operation.
+
+.. note::
+
+    Instance profile credentials and other temporary credentials generated by
+    the AWS Security Token Service (AWS STS) are not supported by every
+    service. Please check if the service you are using supports temporary
+    credentials by reading `AWS Services that Support AWS STS <http://docs.aws.amazon.com/STS/latest/UsingSTS/UsingTokens.html>`_.
+
+For more information, see `IAM Roles for Amazon EC2 Container Service Tasks <http://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-iam-roles.html>`_.
+
+
+.. _assume_role_credentials:
+
+Using Assume Role Credentials
+-----------------------------
+
+Using ``Aws\Credentials\AssumeRoleCredentialProvider`` to create credentials by assuming a role,
+you would need to provide ``'region'`` information, ``'credentials'`` for a ``StsClient`` and
+``'assume_role_params'`` information. Or you could simply pass in ``'client'`` parameter with a ``StsClient``.
+
+For more information regarding ``'assume_role_params'``, see `AssumeRole <http://docs.aws.amazon.com/aws-sdk-php/v3/api/api-sts-2011-06-15.html#assumerole>`_.
+
+.. code-block:: php
+
+    $assumeRoleCredentials = new AssumeRoleCredentialProvider([
+        'region' => 'us-west-2',
+        'credentials' => CredentialProvider::ini(),
```

Comment:
I'm in favored of having a `client` passed in actually, having `credentials` and `region` makes it appears more convenient at first glance but actually introduces more constraints. I'll change that :)